Copy GitHub Pages symbolic links

ashenm opened the pull request:
Instead ignoring symbolic links while deploying to GitHub Pages, the PR proposes to include symbolic links on deploy, which particularly benefit builds that bypasses Jekyll via `.nojekyll`.

ashenm commented:
https://github.com/travis-ci/dpl/issues/912

BanzaiMan commented:
Do you have a build that shows the problem and another that shows how this PR fixes it? Thanks.

ashenm commented:
The build log [here](https://travis-ci.org/ashenm/S3Vault/builds/531172185#L244-L249) as well as the following reproduce evince the exclusion of symbolic link `404.html` concomitant of the current version of the `rsync` [call](https://github.com/travis-ci/dpl/blob/master/lib/dpl/provider/pages.rb#L199).

``` shell
ubuntu@64a9ce222102:~/workspace/S3Vault$ ls -Al dist/
total 360
lrwxrwxrwx 1 ubuntu ubuntu     10 May 30 06:17 404.html -> vault.html
-rw-r--r-- 1 ubuntu ubuntu  25783 May 30 06:17 amazon-cognito-auth.min.js
-rw-r--r-- 1 ubuntu ubuntu 321503 May 30 06:17 aws-sdk-2.384.0.min.js
-rw-r--r-- 1 ubuntu ubuntu      0 May 30 06:17 .nojekyll
-rw------- 1 ubuntu ubuntu    299 May 30 06:25 vault.css
-rw------- 1 ubuntu ubuntu   1604 May 30 06:25 vault.html
-rw------- 1 ubuntu ubuntu   7413 May 30 06:25 vault.min.js
```

``` shell
ubuntu@64a9ce222102:~/workspace/S3Vault$ dpl --skip_cleanup=true --provider=pages --target-branch=assess-dpl --local-dir=dist --github-token="$GITHUB_TOKEN"
Installing deploy dependencies
Preparing deploy
Logged in as @ashenm (Ashen Gunaratne)
Deploying application
cd /tmp/d20190530-2093-1qmksbb/work
Initialized empty Git repository in /tmp/d20190530-2093-1qmksbb/work/.git/
Switched to a new branch 'assess-dpl'
cd -
cd /tmp/d20190530-2093-1qmksbb/work
skipping non-regular file "404.html"
commit 3663fff7a3663b2326684a8b49709ddd758e1241 (HEAD -> assess-dpl)
Author: Deployment Bot (from Travis CI) <deploy@travis-ci.org>
Date:   Thu May 30 06:44:29 2019 +0000

    Deploy  to github.com/.git:assess-dpl

 .nojekyll                  |  0
 amazon-cognito-auth.min.js | 59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 aws-sdk-2.384.0.min.js     |  9 +++++++++
 vault.css                  |  1 +
 vault.html                 |  1 +
 vault.min.js               | 14 ++++++++++++++
 6 files changed, 84 insertions(+)
Couldn't push the build to github.com/.git:assess-dpl
```
The alteration of `rsync` [call](https://github.com/travis-ci/dpl/blob/a74b300872e1d85b87a2ca5b0578736e017cc2af/lib/dpl/provider/pages.rb#L199) to utilise the flag `-l` proposed by the PR yields the following facsimile which divulge the inclusion of symlink `404.html`

```shell
ubuntu@64a9ce222102:~/workspace/S3Vault$ dpl --skip_cleanup=true --provider=pages --target-branch=assess-dpl --local-dir=dist --github-token="$GITHUB_TOKEN"
Installing deploy dependencies
Preparing deploy
Logged in as @ashenm (Ashen Gunaratne)
Deploying application
cd /tmp/d20190530-2131-1b1wsff/work
Initialized empty Git repository in /tmp/d20190530-2131-1b1wsff/work/.git/
Switched to a new branch 'assess-dpl'
cd -
cd /tmp/d20190530-2131-1b1wsff/work
commit f00e620a5039579b1ba60c1a4f006dad44884fa8 (HEAD -> assess-dpl)
Author: Deployment Bot (from Travis CI) <deploy@travis-ci.org>
Date:   Thu May 30 06:45:13 2019 +0000

    Deploy  to github.com/.git:assess-dpl

 .nojekyll                  |  0
 404.html                   |  1 +
 amazon-cognito-auth.min.js | 59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 aws-sdk-2.384.0.min.js     |  9 +++++++++
 vault.css                  |  1 +
 vault.html                 |  1 +
 vault.min.js               | 14 ++++++++++++++
 7 files changed, 85 insertions(+)
Couldn't push the build to github.com/.git:assess-dpl
```
